src/gobupload/storage/handler.py

```diff
@@ -412,52 +465,83 @@ def delete_confirms(self):
 
     @with_session
-    def get_entity_max_eventid(self):
+    def get_entity_max_eventid(self) -> int:
         """Get the highest last_event property of entity
 
         :return: The highest last_event
         """
-        result = self.session.query(self.DbEntity)\
-                     .filter_by(_source=self.metadata.source)\
-                     .order_by(self.DbEntity._last_event.desc())\
-                     .first()
-        return None if result is None else result._last_event
+        last_event = getattr(self.DbEntity, "_last_event")
+        query = (
+            select(last_event)
+            .where(self.DbEntity._source == self.metadata.source)
+            .order_by(last_event.desc())
+            .limit(1)
+        )
+        return self.session.execute(query).scalar() or 0
 
     @with_session
-    def get_last_eventid(self):
+    def get_last_eventid(self) -> int:
         """Get the highest last_event property of entity
 
         :return: The highest last_event
         """
-        result = self.session.query(self.DbEvent) \
-            .filter_by(source=self.metadata.source, catalogue=self.metadata.catalogue, entity=self.metadata.entity) \
-            .order_by(self.DbEvent.eventid.desc())\
-            .first()
-        return None if result is None else result.eventid
+        events = self.DbEvent
+        query = (
+            select(events.eventid)
+            .where(events.source == self.metadata.source)
+            .where(events.catalogue == self.metadata.catalogue)
+            .where(events.entity == self.metadata.entity)
+            .order_by(events.eventid.desc())
+            .limit(1)
+        )
+        return self.session.execute(query).scalar() or 0
 
     @with_session
-    def get_events_starting_after(self, eventid, count=10000):
-        """Return a list of events with eventid starting at eventid
-
-        :return: The list of events
-        """
-        return self.session.query(self.DbEvent) \
-            .filter_by(source=self.metadata.source, catalogue=self.metadata.catalogue, entity=self.metadata.entity) \
-            .filter(self.DbEvent.eventid > eventid if eventid else True) \
-            .order_by(self.DbEvent.eventid.asc()) \
-            .limit(count) \
-            .all()
+    def get_events_starting_after(self, eventid: int, count: int = 10_000) -> Iterator:
+        """
+        Return an iterator of events with eventid starting at eventid.
+        Requires a session in transaction.
+
+        :param eventid: minimal eventid (0 for all)
+        :param count: number of events
+        :return: Iterator of event ORM entities
+        """
+        events = self.DbEvent
+        query = (
+            select(
+                events.eventid,
+                events.timestamp,
+                events.catalogue,
+                events.entity,
+                events.version,
+                events.action,
+                events.source,
+                # events.source_id,
```

Comment:
Why is this commented out?

Reply:
This attribute is not used in the following code, but is part of the events table. Could just remove source_id (is legacy anyway I think)?